src/main/java/frc/robot/LimitedMotor.java

```diff
@@ -0,0 +1,90 @@
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkMaxRelativeEncoder;
+
+class LimitedMotor {
+    // the motor being limited
+    private CANSparkMax spark;
+    // the encoder that belongs to the above spark
+    private RelativeEncoder encoder;
+
+    // the position of the encoder in rotations when it hits the reverse limit switch.
+    private double start;
+    // the distance in rotations from the forward to the reverse limit switch.
+    private final double range;
+    // the maximum speed, in RPM, at which it is safe to run into the limit switch
+    private final double safeSpeed;
+
+    // Tells us whether we have found where the reverse limit switch is
+    private boolean startIsKnown = false;
+
+    // where we want it to go to
+    // Where 1 is the forward limit switch and 0 is the reverse limit switch
+    private double targetPos;
+
+    /**
+     * Creates a new limited motor
+     * 
+     * @param spark the motor to be limited
+     * @param range the distance in rotations from the forward to the reverse limit switch
+     * @param safeSpeed the maximum speed, in RPM, at which it is safe to run into the limit switch
+     */
+    public LimitedMotor(CANSparkMax spark, double range, double safeSpeed) {
+        this.spark = spark;
+        this.range = range;
+        this.safeSpeed = safeSpeed;
+        // 4096 is the ticks per revolution of the encoder
+        this.encoder = spark.getEncoder(SparkMaxRelativeEncoder.Type.kQuadrature, 4096);
```

Comment:
As discussed, we'll likely want to move the 4096 into a constant or parameter to the class.